### PR TITLE
Feat/464: Bypass Cognito Login UI

### PR DESCRIPTION
**Pull request by MCatherine1994**
Bypass cognito login UI

### CODE REVIEW

File: frontend/src/services/AuthService.ts [hunk 1]
```diff
@@ -34,7 +34,15 @@ async function login() {
...
         https://docs.amplify.aws/lib/auth/advanced/q/platform/js/#identity-pool-federation
     */
-    Auth.federatedSignIn();
+    // Auth.federatedSignIn();
+
+    const env = JSON.parse(window.localStorage.getItem('env_data') || '');
+
+    window.location.href =
+        `https://${env?.fam_cognito_domain.value}.auth.${env?.fam_cognito_region.value}.amazoncognito.com/` +
+        `oauth2/authorize?identity_provider=${env?.target_env.value.toUpperCase()}-IDIR&` +
```
Comment (MCatherine1994) [inline frontend/src/services/AuthService.ts:43]: The target_env variable in deployment is actually an injected variable, in the `infrastructure/server/outputs.tf` file, it says as the following, so I assume its value will be "dev" or "test" or "prod"? If so, I just make it to upper case and the identity provider will be "DEV-IDIR" or "TEST-IDIR" or "PROD-IDIR"
```
output "target_env" {
  description = "dev, test, or prod in AWS"
  value = var.target_env
}
```

Comment (basilv) [reply frontend/src/services/AuthService.ts:43]: Hmm, this is only temporary at best until BCeID users are added to FAM. And this solution seems like a bit of a hack.
For example, assuming the identity provider name could be avoiding by passing in the IDIR identity provider name into the env_data.

File: frontend/src/services/AuthService.ts [hunk 2]
```diff
@@ -34,7 +34,15 @@ async function login() {
         https://docs.amplify.aws/lib/auth/social/q/platform/js/
         https://docs.amplify.aws/lib/auth/advanced/q/platform/js/#identity-pool-federation
     */
-    Auth.federatedSignIn();
+    // Auth.federatedSignIn();
+
+    const env = JSON.parse(window.localStorage.getItem('env_data') || '');
```
Comment (basilv) [inline frontend/src/services/AuthService.ts:39]: The EnvironmentSettings class should be used instead as that abstracts accessing environment-specific data.

File: frontend/src/services/AuthService.ts [hunk 3]
```diff
@@ -34,7 +34,15 @@ async function login() {
         https://docs.amplify.aws/lib/auth/social/q/platform/js/
         https://docs.amplify.aws/lib/auth/advanced/q/platform/js/#identity-pool-federation
     */
-    Auth.federatedSignIn();
```
Comment (basilv) [inline frontend/src/services/AuthService.ts:37]: Not using Auth.federatedSignIn seems to bypass the Amplify library entirely, at least for login. Is there no way to pass in the identityProvider option into the signin method? Documentation suggests this is possible - e.g. 
After configuring the OAuth endpoints (Cognito Hosted UI), you can integrate your App by invoking Auth.federatedSignIn() function. Passing LoginWithAmazon, Facebook, Google, or SignInWithApple on the provider argument (e.g Auth.federatedSignIn({ provider: 'LoginWithAmazon' }))

**Review by basilv — CHANGES_REQUESTED** [review]
@MCatherine1994 Review done, changes suggested.

File: frontend/src/services/AuthService.ts (hunk 3 above)
Comment (MCatherine1994) [reply frontend/src/services/AuthService.ts:37]: Updated to use customProvider option as mentioned in this example: 
https://github.com/aws-amplify/amplify-js/issues/10900

File: frontend/public/env.json [hunk 4]
```diff
@@ -29,6 +29,11 @@
         "type": "string",
         "value": "http://localhost:5173"
     },
+    "fam_console_idp_name": {
+        "sensitive": false,
```
Comment (MCatherine1994) [inline frontend/public/env.json:33]: This is an existing variable in the terraform

**Review by basilv — APPROVED** [review]
I'm fine with this. I'm also fine if you want to try moving the customProvider setting into the AWS Config as per Ian's suggestion.

**Comment by MCatherine1994** [comment]
> I'm fine with this. I'm also fine if you want to try moving the customProvider setting into the AWS Config as per Ian's suggestion.

I didn't find a solution for setup that in the Amlify config method. I checked in their GitHub repos issues, most of the people are using the customProvider in the signin method to bypass the Cognito UI.

**Pull request merged**